Modules/socketmodule.c

```diff
@@ -6175,15 +6153,10 @@ socket_dup(PyObject *self, PyObject *fdobj)
 
     newfd = WSASocketW(FROM_PROTOCOL_INFO, FROM_PROTOCOL_INFO,
                       FROM_PROTOCOL_INFO,
-                      &info, 0, WSA_FLAG_OVERLAPPED);
+                      &info, 0,
+                      WSA_FLAG_OVERLAPPED | WSA_FLAG_NO_HANDLE_INHERIT);
     if (newfd == INVALID_SOCKET)
         return set_error();
-
-    if (!SetHandleInformation((HANDLE)newfd, HANDLE_FLAG_INHERIT, 0)) {
-        closesocket(newfd);
-        PyErr_SetFromWindowsErr(0);
-        return NULL;
-    }
```

Comment:
@eryksun are you sure this is the correct change? This seems to break the tests.

Reply:
Sorry. I forgot that the `WSASocketW()` call in this case isn't responsible for duplicating the handle into the current process. It's implemented the other way around. The API can't assume that the process that calls `WSASocketW()` is allowed to duplicate a handle from the process that calls `WSADuplicateSocketW()`.  Instead the value of the already duplicated handle is passed in the `WSAPROTOCOL_INFOW` record. This handle is inheritable, so calling `SetHandleInheritable()` is required in this case.

Reply:
Calling `SetHandleInformation()` is thus also required in `sock_initobj_impl()` for the case where a socket is created with `FROM_PROTOCOL_INFO`. I'd do the `SetHandleInformation()` call locally right after calling `WSASocketW()`, and add a comment that explains why `WSA_FLAG_NO_HANDLE_INHERIT` doesn't work.

The other `WSASocketW()` call in `sock_initobj_impl()` doesn't need `SetHandleInformation()`. Keeping it separate and adding a comment makes it less likely that we'll mistakenly go down this path again.